R/middleware.R

```
@@ -299,18 +299,37 @@ HandlerManager <- R6Class("HandlerManager",
 
           if (reqSize > maxSize) {
             return(list(status = 413L,
-              headers = list(
-                'Content-Type' = 'text/plain'
-              ),
+              headers = list('Content-Type' = 'text/plain'),
               body = 'Maximum upload size exceeded'))
           }
           else {
             return(NULL)
           }
         },
         call = .httpServer(
-          function (req) {
-            withLogErrors(handlers$invoke(req))
+          function(req,
+                   full = getOption("shiny.fullstacktrace", FALSE),
+                   offset = getOption("shiny.stacktraceoffset", TRUE)) {
+
+             withCallingHandlers(captureStackTraces(handlers$invoke(req)),
+               error = function(cond) {
+                 # Don't print shiny.silent.error (i.e. validation errors)
+                 if (inherits(cond, "shiny.silent.error")) return()
+                 if (isTRUE(getOption("show.error.messages"))) {
+                   printError(cond, full = full, offset = offset)
+                   sanitizeErrors = getOption('shiny.sanitize.errors', TRUE)
+
+                   if (inherits(cond, 'shiny.custom.error') || !sanitizeErrors) {
+                     stop(cond$message, call. = FALSE)
```

Well, I do want to throw the error at some point (because I want an error message to show up) - right? If so, I think this is the place to do it...